supervisor_client.py

```diff
@@ -0,0 +1,28 @@
+import http.client
```

Comment:
The code doesn't come from any project, it's my own adaptation of a fix that's been circulating [on the stacks](https://stackoverflow.com/a/23837147).

I only had to change a few imports to make the transition to python3.